File: KeeperNg/src/app/client/feature/client-repository/feature/client-repository-files/model/repository.model.ts
import { RepoInterface } from "src/app/shared/interfaces/repo.interface";

export class RepositoryModel implements RepoInterface {
    id: string;
    name: string;
    description: string;
    createdDate: string;
    overallFileCount: number;
    overallRepositorySize: number;

    constructor(repo: RepoInterface) {
        this.id = repo.id;
        this.name = repo.name;
        this.description = repo.description;
        this.createdDate = repo.createdDate;
        this.overallFileCount = repo.overallFileCount;
        this.overallRepositorySize = repo.overallRepositorySize;
    }
    
    
}

File: src/Keeper.Domain/Models/RepositoryModel.cs
namespace Keeper.Domain.Models;

public class RepositoryModel
{
    public Guid Id { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public bool AllowAnonymousFileRead { get; set; }
    public long OverallFileCount { get; set; }
    public long OverallRepositorySize { get; set; }
    public DateTime CreatedDate { get; set; }
}
